helix-term/src/application.rs

```diff
@@ -52,77 +52,110 @@ pub struct Application {
 impl Application {
     pub fn new(args: Args, mut config: Config) -> Result<Self, Error> {
         use helix_view::editor::Action;
-        let mut compositor = Compositor::new()?;
-        let size = compositor.size();
-
-        let conf_dir = helix_core::config_dir();
 
-        let theme_loader =
-            std::sync::Arc::new(theme::Loader::new(&conf_dir, &helix_core::runtime_dir()));
-
-        // load default and user config, and merge both
-        let builtin_err_msg =
-            "Could not parse built-in languages.toml, something must be very wrong";
-        let def_lang_conf: toml::Value =
-            toml::from_slice(include_bytes!("../../languages.toml")).expect(builtin_err_msg);
-        let def_syn_loader_conf: helix_core::syntax::Configuration =
-            def_lang_conf.clone().try_into().expect(builtin_err_msg);
-        let user_lang_conf = std::fs::read(conf_dir.join("languages.toml"))
-            .ok()
-            .map(|raw| toml::from_slice(&raw));
-        let lang_conf = match user_lang_conf {
-            Some(Ok(value)) => Ok(merge_toml_values(def_lang_conf, value)),
-            Some(err @ Err(_)) => err,
-            None => Ok(def_lang_conf),
-        };
-
-        let theme = if let Some(theme) = &config.theme {
-            match theme_loader.load(theme) {
-                Ok(theme) => theme,
-                Err(e) => {
-                    log::warn!("failed to load theme `{}` - {}", theme, e);
-                    theme_loader.default()
-                }
-            }
-        } else {
-            theme_loader.default()
-        };
+        // These configuration directories can contain `config.toml` and `languages.toml`.
+        // `local_config_dir` is a `.helix` folder within the projec directory.
+        let config_dir = helix_core::config_dir();
+        let local_config_dir = helix_core::local_config_dir();
+
+        // Config override order: local -> global -> default.
```

Comment:
```suggestion
        // Config override order: local (cwd) -> local (root) -> global -> default.
```
I think this is a better option.